File: src-tauri/src/models/statistics.model.rs
/* sys lib */
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[allow(non_snake_case)]
pub struct StatisticsModel {
  pub totalTasks: i32,
  pub completionRate: i32,
  pub averageTaskTime: i32,
  pub productivityScore: i32,
  pub previousTotalTasks: i32,
  pub previousCompletionRate: i32,
  pub previousAverageTime: i32,
  pub previousProductivityScore: i32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[allow(non_snake_case)]
pub struct CompletionTrendItem {
  pub label: String,
  pub value: i32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[allow(non_snake_case)]
pub struct DailyActivityItem {
  pub dayName: String,
  pub activity: i32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[allow(non_snake_case)]
pub struct CategoryItem {
  pub name: String,
  pub count: i32,
  pub percentage: i32,
  pub color: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[allow(non_snake_case)]
pub struct ChartDataModel {
  pub completionTrend: Vec<CompletionTrendItem>,
  pub categories: Vec<CategoryItem>,
  pub dailyActivity: Vec<DailyActivityItem>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[allow(non_snake_case)]
pub struct AchievementModel {
  pub title: String,
  pub description: String,
  pub icon: String,
  pub color: String,
  pub date: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[allow(non_snake_case)]
pub struct DetailedMetricModel {
  pub name: String,
  pub current: String,
  pub previous: String,
  pub change: i32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[allow(non_snake_case)]
pub struct StatisticsResponseModel {
  pub statistics: StatisticsModel,
  pub chartData: ChartDataModel,
  pub achievements: Vec<AchievementModel>,
  pub detailedMetrics: Vec<DetailedMetricModel>,
}

File: src/app/models/statistics.model.ts
export interface Statistics {
  total_tasks: number;
  completion_rate: number;
  average_task_time: number;
  productivity_score: number;
  previous_total_tasks: number;
  previous_completion_rate: number;
  previous_average_time: number;
  previous_productivity_score: number;
}

export interface StatisticsResponse {
  statistics: Statistics;
  chart_data: ChartData;
  achievements: Achievement[];
  detailed_metrics: DetailedMetric[];
}

export interface ChartData {
  completion_trend: Array<{ label: string; value: number }>;
  categories: Array<CategoryChart>;
  daily_activity: Array<{ day_name: string; activity: number }>;
}

export interface CategoryChart {
  name: string;
  count: number;
  percentage: number;
  color: string;
}

export interface Achievement {
  title: string;
  description: string;
  icon: string;
  color: string;
  date: string;
}

export interface DetailedMetric {
  name: string;
  current: string;
  previous: string;
  change: number;
}
